http/http.go

```diff
@@ -135,14 +135,43 @@ func (s Server) viewPost(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func (s Server) editPost(w http.ResponseWriter, r *http.Request) {
+	ID, err := strconv.ParseFloat(r.PathValue("id"), 32)
+	if err != nil {
+		log.Fatalf("convert to float: %v", err)
+	}
+
+	var p db.Post
+
+	date, err := strconv.ParseFloat(r.FormValue("date"), 32)
+	if err != nil {
+		log.Fatalf("convert to float: %v", err)
+	}
+
+	p.ID = float32(ID)
+	p.Date = float32(date)
+	p.Title = r.FormValue("title")
+	p.Link = r.FormValue("link")
+	p.Content = r.FormValue("content")
```

Comment:
Most of this is repeated from `postAPIPosts`, so maybe you can create a function that parses posts form values?